### PR TITLE
junit - handle tests without names

### PR DESCRIPTION
**Pull request by jeremylt**
The extra , in TAP mode when the test doesn't have a name was driving me a little batty

### CODE REVIEW

File: tests/junit_common.py [hunk 1]
```diff
@@ -384,8 +385,9 @@ def run_test(index: int, test: str, spec: TestSpec, backend: str,
 
     # run test
     skip_reason: str = suite_spec.check_pre_skip(test, spec, backend, nproc)
+    test_name_str: str = f'"{spec.name}", ' if spec.name else ''
```
Comment (jedbrown) [inline tests/junit_common.py:388]: Should we require that all tests be named (many testing frameworks do), or perhaps automatically assign them names?

Comment (jeremylt) [reply tests/junit_common.py:388]: Adding names would be a lot of changes to almost every test file. I'm not opposed, but its also another thing to manually change every time we make a new test. I could go either way

Comment (jedbrown) [reply tests/junit_common.py:388]: It looks like every file with more than one test names them. We could assert that (if you have multiple tests they all need names), but I feel like the sed operation to add `name="default",` to all the unnamed ones in `t*.c` files is also fine. One thing I've considered is using libtest-mimic to support using `cargo test` to run these tests, and if I do that, I'll need to name them all (either explicitly or programmatically).

Comment (zatkins-dev) [reply tests/junit_common.py:388]: I'm good with either approach

**Pull request closed without merging**